test/e2e/framework/gateways.go

```diff
@@ -184,30 +183,18 @@ func (f *Framework) DeleteGateway(cluster ClusterIndex, name string) {
 	}, NoopCheckResult)
 }
 
-// GatewayCleanup ensures that only the active gateway node is flagged as gateway node which could not be after a failed
-//  test which left the system on an unexpected state.
-func (f *Framework) GatewayCleanup() {
+// KindGatewayCleanup will be executed only on kind environment.
+// It will restore the gateway nodes to its initial state.
+// Other environments do not need any gw cleanup as MachineSet is responsible to keeping the gw nodes in active states.
+func (f *Framework) KindGatewayCleanup() {
```

Comment:
Minor thing but I think we can keep the function name as is since it internally checks if the environment is "kind". Having `Kind` in the function name implies the caller should ensure the environment is "kind" but that's not the case.

Reply:
ack, makes sense.
Will change the name back.